### PR TITLE
Trial stale issue and PR action

### PR DESCRIPTION
**Pull request by ChrisChinchilla**
### Description

Figured it might be worth testing this action to see how well it helps us identify old issues and PRs, we can disable it if it's annoying.

### Checklist
- [x] Code compiles correctly
- [x] All tests are passing
- [x] New tests have been created which fail without the change (if possible)
- [x] README / documentation was extended, if necessary
- [x] Changelog entry (if change is visible to the user)
- [x] Used meaningful commit messages

### CODE REVIEW

File: .github/workflows/stale.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+jobs:
+  stale:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/stale@v1
+      with:
+        repo-token: ${{ secrets.GITHUB_TOKEN }}
+        stale-issue-message: 'Stale issue message'
```
Comment (Marenz) [inline .github/workflows/stale.yml]: Shouldn't this (and the line below) be the actual message that will be posted to stale issues/prs?

Comment (ChrisChinchilla) [reply .github/workflows/stale.yml]: TBH, I just added the default action template, so yes you're right

```suggestion
        stale-issue-message: 'Potential stale issue'
```

Or something like that?

File: .github/workflows/stale.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        stale-pr-message: 'Stale pull request message'
```
Comment (ChrisChinchilla) [inline .github/workflows/stale.yml]:
```suggestion
        stale-pr-message: 'Stale pull request'
```

Again, something like that

**Comment by chriseth** [comment]
How are stale issues and stale pull requests defined?

**Comment by bshastry** [comment]
> How are stale issues and stale pull requests defined?

tl;dr we define how many days after inactivity, issue/PR becomes stale

FWIW: If the underlying script used by this github action is [this one](https://github.com/actions/stale), then we need to define the fields marked `required: true` here https://github.com/actions/stale/blob/9f69ef12ad1ae539589cc84100eacea217506838/src/main.ts#L147-L165


Edit: I'm actually not sure if all required fields should be set. `daysBeforeClose` for example, automatically closes an issue that has remained inactive for `daysBeforeClose` days. I mean, do we want that to happen?

**Comment by chriseth** [comment]
I don't think issues should be considered stale.

**Comment by ChrisChinchilla** [comment]
Try these changes

File: .github/workflows/stale.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+jobs:
+  stale:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/stale@v1
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+          stale-pr-message: "A reminder that this pull request has had no activity for 30 days"
+          stale-pr-label: "no-pr-activity"
```
Comment (Marenz) [inline .github/workflows/stale.yml:16]: I suppose we'll need to create that label, too

**Review by Marenz — DISMISSED** [review]

File: .github/workflows/stale.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+          days-before-stale: 30
```
Comment (ekpyron) [inline .github/workflows/stale.yml]:
```suggestion
          days-before-stale: 14
```
:-)?

Comment (ChrisChinchilla) [reply .github/workflows/stale.yml]: You're keen ;)

I really don't mind, but that seems pretty quick? 21? Meet in the middle?

Comment (ekpyron) [reply .github/workflows/stale.yml]: I don't really mind either, so as you like - I just thought 30 days is actually a lot of inactivity.

**Comment by ChrisChinchilla** [comment]
@ekpyron Accepted change, forces us to be proactive :)

**Comment by ekpyron** [comment]
The PR needs to be rebased to current develop, though (we changed the testing docker image and adjusted the tests on top of that, that's why the tests fail here now...)

**Comment by chriseth** [comment]
Or we just ignore the tests because there is nothing to be tested.

**Comment by Marenz** [comment]
Well, at least I can't merge (button greyed out), supposedly because of the failing tests.

**Comment by ChrisChinchilla** [comment]
Of course we changed the tests 😆

Rebased, let's see how this fixes things.

**Comment by Marenz** [comment]
squash commits?

**Comment by ChrisChinchilla** [comment]
@Marenz Ugh, too may things to remember 😩

Done…

File: .github/workflows/stale.yml [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+jobs:
+  stale:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/stale@v1
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+          stale-pr-message: "A reminder that this pull request has had no activity for 30 days"
```
Comment (chriseth) [inline .github/workflows/stale.yml]:
```suggestion
          stale-pr-message: "A reminder that this pull request has had no activity for 14 days"
```

File: .github/workflows/stale.yml (hunk 3 above)
Comment (ChrisChinchilla) [reply .github/workflows/stale.yml:16]: I think it does it automatically?

**Review by chriseth — APPROVED** [review]
